hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -303,15 +304,17 @@ public void start() {
           datanodeDetails.setPort(DatanodeDetails.newPort(HTTPS,
                   httpServer.getHttpsAddress().getPort()));
         }
+        serviceRuntimeInfo.setHttpPort(String.valueOf(httpServer.getHttpAddress().getPort()));
```

Comment:
What about the HTTPS port and address? are other components publishing these as separate metrics?

Reply:
You are correct. we should add the corresponding port metric for `HTTPS`. As for the `hostname`, I believe we can have `HTTP` and `HTTPS` share the same one, since they are theoretically consistent.